server/rest/src/main/java/org/infinispan/rest/resources/ServerHealthCheckResource.java

```diff
@@ -0,0 +1,74 @@
+package org.infinispan.rest.resources;
+
+import static org.infinispan.rest.framework.Method.GET;
+import static org.infinispan.rest.framework.Method.HEAD;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.commons.CacheException;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.factories.GlobalComponentRegistry;
+import org.infinispan.lifecycle.ComponentStatus;
+import org.infinispan.manager.DefaultCacheManager;
+import org.infinispan.rest.InvocationHelper;
+import org.infinispan.rest.NettyRestResponse;
+import org.infinispan.rest.framework.ResourceHandler;
+import org.infinispan.rest.framework.RestRequest;
+import org.infinispan.rest.framework.RestResponse;
+import org.infinispan.rest.framework.impl.Invocations;
+import org.infinispan.security.actions.SecurityActions;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+public class ServerHealthCheckResource implements ResourceHandler {
+
+   private final InvocationHelper helper;
+
+   public ServerHealthCheckResource(InvocationHelper helper) {
+      this.helper = helper;
+   }
+
+   @Override
+   public Invocations getInvocations() {
+      return new Invocations.Builder()
+
+            .invocation().methods(GET, HEAD).path("/health/live")
```

Comment:
We should set `anonymous(true)` for both endpoints, as this allows us to avoid configuring auth in the StatefulSet.